include/chemfiles/selections/expr.hpp

```diff
@@ -125,6 +136,10 @@ class SubSelection {
     std::unique_ptr<Selection> selection_;
     /// Variable to use if selection_ is nullptr
     Variable variable_;
+    /// Cache matches for the selection on the first call to eval
+    mutable std::vector<size_t> matches_;
```

Comment:
Why is this marked mutable? Is it to be modified by a `const` function?

Reply:
Yes. `Selector::is_match` is marked as const because it should behave as a const function for any external observer. Having a cache should not change this, and thus the modified fields needs to be marked as `mutable`